ClassicLootManager/Modules/Rosters/RosterManager/ExpressionParser.lua

```diff
@@ -0,0 +1,425 @@
+        ["+"] = false,
+        ["+a"] = false,
+        ["a+"] = false,
+        ["-"] = false,
+        ["-a"] = true,
+        ["--a"] = false,
+        ["---a"] = false,
+        ["a-"] = false,
+        ["/"] = false,
+        ["/a"] = false,
+        ["a/"] = false,
+        ["*"] = false,
+        ["*a"] = false,
+        ["a*"] = false,
+        ["^"] = false,
+        ["^a"] = false,
+        ["a^"] = false,
+
+        -- Repeating operator
+        ["a++b"] = false,
+        ["a--b"] = true,
+        ["a---b"] = false,
+        ["a+--b"] = false,
+        ["a--+b"] = false,
+        ["a//b"] = false,
+        ["a**b"] = false,
+        ["a^^b"] = false,
+        ["a+-b"] = true,
+        ["a^-b"] = true,
+        ["a*-b"] = true,
+        ["a/-b"] = true,
+        ["a-/b"] = false,
+        ["a/*b"] = false,
+        ["a*^b"] = false,
+        ["a^+b"] = false,
+
+        -- operator before closing parenthesis
+        ["(a+)"] = false,
+        ["(a-)"] = false,
+        ["(a/)"] = false,
+        ["(a*)"] = false,
+        ["(a^)"] = false,
+
+        -- empty parenthesis
+        ["()"] = false,
+        ["a^()"] = false,
+        ["ab+()"] = false,
+        ["(a)"] = true,
+        ["(a+b)"] = true,
+
+        -- non-operator before opening parenthesis
+        ["a(b)"] = false,
+        ["1(1)"] = false,
+        ["a+(b)"] = true,
+        ["1+(1)"] = true,
+
+        -- text
+        ["abc"] = false,
+        ["ab+c"] = false,
+        ["c"] = false,
+        ["ab"] = true,
+        ["a+b"] = true,
+        ["a"] = true,
+
+        -- parenthesis balancing
+        [")("] = false,
+        ["(a"] = false,
+        ["a)"] = false,
+        ["(a))"] = false,
+        ["((a)"] = false,
+        ["((a))"] = true,
+
+        -- special characters
+        ["d_e"] = false,
+        ["§"] = false,
+
+        -- empty expression
+        [""] = false,
+
+        -- floating point
+        ["1.1"] = true,
+        ["1.1+1.1"] = true,
+        ["1.1+(1.1)"] = true,
+        ["(1.1)"] = true,
+        ["(1.1+1.1)"] = true,
+        ["(1.1+(1.1))"] = true,
+        [".1"] = false,
+        ["1."] = false,
+        ["."] = false,
+        ["1.1."] = false,
+        ["1.1.1"] = false,
+        ["1.1.1+1.1.1"] = false,
+        ["1.1.1+(1.1.1)"] = false,
+        ["(1.1.1)"] = false,
+        ["(1.1.1+1.1.1)"] = false,
+        ["(1.1.1+(1.1.1))"] = false,
+
+        -- negative numbers
+        ["-1"] = true,
+        ["1--1"] = true,
+        ["1*-1"] = true,
+        ["1^-1"] = true,
+        ["1/-1"] = true,
+        ["1+-1"] = true,
+        ["(-1)"] = true,
+
+        -- random positive cases
+        ["a+b"] = true,
+        ["ab"] = true,
+        ["b^(a)"] = true,
+        ["b^(1)"] = true,
+        ["10^(1)"] = true,
+        ["((3)/1)"] = true,
+        ["ab+(b)"] = true,
+        ["(a)"] = true,
+        ["-1+1*2^3+4/5-6+(3-4)^((5+6)+(2^(-3+9)))+a"] = true
+    }
+
+    local maxTestLength = 0
+    for test, _ in pairs(tests) do
```

Comment:
why do you set true/false (or even use hash table instead of list) when you dont even check the value?

Reply:
The test setup was very rushed and kinda poor. It could use cleanup, but I cant justify spending more time on it. This part is actually just padding for the header. The part that evaulates the test is just a bit further down.  The boolean is the expected result.

Reply:
It also doesn't work very well with the CLM constants etc. So I think the tests just have to be dropped entirely, they were useful when I did the development of the solution to verify the behavior. Unless you are aware of a nicer unit testing framework for lua that plays nice with the WoW API